aggregator_core/src/batch_mode.rs

```diff
@@ -346,10 +339,11 @@ impl TimeIntervalBatchIdentifierIter {
         // Sanity check that the given interval is of an appropriate length. We use an assert as
         // this is expected to be checked before this method is used.
         assert_eq!(
-            batch_interval.duration().as_seconds() % time_precision.as_seconds(),
+            batch_interval.duration().as_seconds(time_precision) % time_precision.as_seconds(),
             0
```

Comment:
`batch_interval.duration() == Duration::ZERO`? Or we could define a method `Duration::is_zero` that does the same thing.

Reply:
This whole bit is wack. The assert goes away, and this becomes:
```
impl TimeIntervalBatchIdentifierIter {
    fn new(time_precision: &TimePrecision, batch_interval: &Interval) -> Self {
        Self {
            step: 0,
            total_step_count: batch_interval.duration().as_time_precision_units(),
            start: *batch_interval.start(),
            time_precision: *time_precision,
        }
    }
}
```

Reply:
in dd359ca08df24fbbfabe59bb509f892e646aab53